src/foam/nanos/dig/DUGRuleAction.js

```diff
@@ -41,32 +45,45 @@ foam.CLASS({
     {
       name: 'applyAction',
       javaCode: `
-        agency.submit(x, new ContextAgent() {
-          @Override
-          public void execute(X x) {
-            var pm = new PM(DUGRuleAction.getOwnClassInfo().getId(), dugRule.getId(), dugRule.getName());
-            try {
-              HTTPSink sink = new HTTPSink(
-                dugRule.getUrl(),
-                dugRule.evaluateBearerToken(),
-                dugRule.getFormat(),
-                new foam.lib.AndPropertyPredicate(x,
-                  new foam.lib.PropertyPredicate[] {
-                    new foam.lib.ExternalPropertyPredicate(),
-                    new foam.lib.NetworkPropertyPredicate(),
-                    new foam.lib.PermissionedPropertyPredicate()}),
-                true
-              );
+      final var dugRule = (DUGRule) rule;
+        
+      agency.submit(x, new ContextAgent() {
+        @Override
+        public void execute(X x) {
+          PM pm = new PM(getClass(), rule.getDaoKey() + "/" + rule.getName());
```

Comment:
PM constructors takes a var arg of strings and does it's own concatenation with : 
If that saves you some headache with building a string yourself, and is consistent with other PMs.